.github/workflows/doc-build.yml

```diff
@@ -114,6 +114,18 @@ jobs:
         name: pr_num
         path: pr_num
 
+    # Keep this last not to break upload and other "functional" tests above.
+    - name: time-incremental-build
+      if: github.event_name == 'pull_request'
+      run: |
+        # Catch major incremental build time regressions like the ones
+        # fixed by commits 52842317505f, a140a249b33b and 9d119431945f.
+        sed -i -e 's/Zephyr /Xephyr /' doc/index.rst # make a small .rst change
+        export DOC_TAG=development SPHINXOPTS='-j auto -t publish' # as above
+        time timeout 35 make -C doc/  html-fast    # ${DOC_TARGET} as above
+        git checkout HEAD -- doc/index.rst         # restore to be safe
```

Comment:
I don't see the need to add such hacks in CI. If something breaks, one can open an issue. GH actions also support timeouts.

Reply:
> I don't see the need to add such hacks in CI. If something breaks, one can open an issue. 

The purpose of CI is to catch regressions _when they happen_. It took me a day to isolate the numfig regression. Few people contribute to documentation and from past experience it seems many of them don't notice how slow the incremental build becomes - which I think makes fewer people contribute to documentation.

> GH actions also support timeouts.

I will take a look, thanks.

Reply:
+1

Re: GH Actions, it's not clear if this would allow to set a timeout of less than a minute https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes